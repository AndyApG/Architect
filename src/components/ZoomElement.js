/*eslint-disable*/

import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import anime from 'animejs';
import { animation } from 'network-canvas-ui';

class ZoomElement extends PureComponent {
  static propTypes = {
    children: PropTypes.any.isRequired,
    colors: PropTypes.array,
  };

  static defaultProps = {
    colors: ['#ffffff', '#2d2955'],
  };

  onClick = (e) => {
    const start = this.node.getBoundingClientRect();
    const pseudoElement = document.createElement('div');
    const body = document.getElementsByTagName('body')[0];
    const width = window.innerWidth;
    const height = window.innerHeight;
    const startX = start.left + start.width / 2;
    const startY = start.top + start.height / 2;

    pseudoElement.setAttribute(
      'style',
      `position: absolute; transform-origin: ${startX}px ${startY}px; transform: translateZ(0); top: 0; left: 0; width: ${width}px; height: ${height}px;`,
    );
    body.appendChild(pseudoElement);

    anime({
      targets: pseudoElement,
      elasticity: 0,
      easing: 'easeInOutQuad',
      duration: animation.duration.standard,
      opacity: [1, 1],
      scale: [0, 1],
      backgroundColor: this.props.colors,
    }).finished.then(() => {
      body.removeChild(pseudoElement);
    });
  }

  render() {
    const {
      children,
      ...props
    } = this.props;

    return (
      <div ref={(node) => {this.node = node; }} onClick={this.onClick} {...props} >
        {children}
      </div>
    );
  }
}

export default ZoomElement;
