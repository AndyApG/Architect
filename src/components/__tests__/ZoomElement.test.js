/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import ZoomElement from '../ZoomElement';

const mockProps = {
};

describe('<ZoomElement />', () => {
  it('can render', () => {
    const component = shallow(<ZoomElement {...mockProps} >Foo</ZoomElement>);

    expect(component).toMatchSnapshot();
  });
});
