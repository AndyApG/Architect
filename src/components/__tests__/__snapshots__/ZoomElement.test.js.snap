// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ZoomElement /> can render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ZoomElement
    colors={
        Array [
            "#ffffff",
            "#2d2955",
          ]
    }
>
    Foo
</ZoomElement>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": "Foo",
      "colors": Array [
        "#ffffff",
        "#2d2955",
      ],
      "onClick": [Function],
    },
    "ref": [Function],
    "rendered": "Foo",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": "Foo",
        "colors": Array [
          "#ffffff",
          "#2d2955",
        ],
        "onClick": [Function],
      },
      "ref": [Function],
      "rendered": "Foo",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
